src/rust/ensogl/lib/select/src/component.rs

```diff
@@ -0,0 +1,403 @@
+    model   : Model,
+    pub frp : Frp,
+}
+
+impl Deref for Select {
+    type Target = Frp;
+    fn deref(&self) -> &Self::Target { &self.frp }
+}
+
+impl Select {
+    /// Constructor.
+    pub fn new(app:&Application) -> Self {
+        let frp           = Frp::new_network();
+        let model         = Model::new(app);
+        Select{frp,model}.init(app)
+    }
+
+    fn init(self, app:&Application) -> Self {
+        const MAX_SCROLL:f32 = entry::HEIGHT/2.0;
+
+        let frp              = &self.frp;
+        let network          = &frp.network;
+        let model            = &self.model;
+        let scene            = app.display.scene();
+        let mouse            = &scene.mouse.frp;
+        let window_y         = Animation::<f32>::new(&network);
+        let selection_y      = Animation::<f32>::new(&network);
+        let selection_height = Animation::<f32>::new(&network);
+
+        frp::extend!{ network
+
+            // === Mouse Position ===
+
+            mouse_in <- mouse.position.all_with(&frp.size, f!([model,scene](pos,size) {
+                let pos_obj_space = scene.screen_to_object_space(&model.background,*pos);
+                let x_range = (-size.x / 2.0)..=(size.x / 2.0);
+                let y_range = (-size.y / 2.0)..=(size.y / 2.0);
+                x_range.contains(&pos_obj_space.x) && y_range.contains(&pos_obj_space.y)
+            }));
+            mouse_moved       <- mouse.distance.map(|dist| *dist > std::f32::EPSILON);
+            mouse_y_in_scroll <- mouse.position.map(f!([model,scene](pos) {
+                scene.screen_to_object_space(&model.scrolled_area,*pos).y
+            }));
+            mouse_pointed_entry <- mouse_y_in_scroll.map(f!((y)
+                model.entries.entry_at_y_position(*y).entry())
+            );
+
+
+            // === Selected Entry ===
+            frp.source.selected_entry <+ frp.select_entry.map(|id| Some(*id));
+
+            selection_jump_on_one_up  <- frp.move_selection_up.constant(-1);
+            selection_jump_on_page_up <- frp.move_selection_page_up.map(f!([model](())
+                -(model.entries.visible_entry_count() as isize))
+            );
+            selection_jump_on_one_down <- frp.move_selection_down.constant(1);
+            selection_jump_on_page_down <- frp.move_selection_page_down.map(f!([model](())
+                model.entries.visible_entry_count() as isize)
+            );
+            selection_jump_up   <- any(selection_jump_on_one_up,selection_jump_on_page_up);
+            selection_jump_down <- any(selection_jump_on_one_down,selection_jump_on_page_down);
+            selected_entry_after_jump_up <- selection_jump_up.map2(&frp.selected_entry,
+                |jump,id| {
+                    let id = id.as_ref()?;
+                    if *id == 0 && *jump != 0 { None                                    }
+                    else                      { Some(id.saturating_sub(-jump as usize)) }
+                }
+            );
```

Comment:
this is complex. Should be refactored to model with a good name imo. If not to model, then to a global non-public function with a good name.